client/gui/SDL_Extensions.h

```diff
@@ -152,6 +152,57 @@ struct ColorPutter
 
 typedef void (*BlitterWithRotationVal)(SDL_Surface *src,SDL_Rect srcRect, SDL_Surface * dst, SDL_Rect dstRect, ui8 rotation);
 
+class ColorShifter
+{
+public:
+	virtual SDL_Color shiftColor(SDL_Color clr) = 0;
+};
+
+class ColorShifterLightBlue : public ColorShifter
+{
+public:
+	static ColorShifterLightBlue* create()
+	{
+		return new ColorShifterLightBlue();

```

Comment:
In this case ColorShifter objects will always leak. You could f.e. save local instance with unique_ptr and return pointer to it.

Reply:
Updated with unique_ptr.